src/esp/assets/ResourceManager.cpp

```diff
@@ -2962,6 +2962,16 @@ void ResourceManager::loadTextures(Importer& importer,
           format = Mn::GL::textureFormat(image->compressedFormat());
         } else {
           format = Mn::GL::textureFormat(image->format());
+          // Modify swizzle for single channel textures so that they are
+          // greyscale
+          if (pixelFormatSize(image->format()) == 1) {
```

Comment:
Not the right API, there's also 16-bit formats, half-floats, ... which can all be single-channel too. You're looking for [pixelFormatChannelCount()](https://doc.magnum.graphics/magnum/namespaceMagnum.html#ab6e2600dba325d29743fad354e60b1e2) instead.

Reply:
Oh, and you may want to handle two-channel images as well, which usually represent greyscale + alpha, and thus should be expanded to RRRG.

Reply:
Thanks!